### ISSUE TITLE
`cfgparser.optionxform = str` gives erroneous error

### ISSUE THREAD

**Issue by adamtheturtle** [python/mypy#5062]
The example at https://docs.python.org/3.6/library/configparser.html#configparser.ConfigParser.optionxform on hpw to make `ConfigParser` option names gives a `mypy` error.

I originally created a typeshed issue at https://github.com/python/typeshed/issues/1857 and @JelleZijlstra suggested that this is instead a `mypy` bug.

* Are you reporting a bug, or opening a feature request?

A bug.
* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python
from configparser import ConfigParser

cfgparser = ConfigParser()
cfgparser.optionxform = str
```

* What is the actual behavior/output?

```
example.py:4: error: Cannot assign to a method
example.py:4: error: Incompatible types in assignment (expression has type "Type[str]", variable has type "Callable[[str], str]")
```

* What is the behavior/output you expect?

I expect no error in this case.

* What are the versions of mypy and Python you are using?

mypy 0.600
Python 3.6.5

* What are the mypy flags you are using? (For example --strict-optional)

Just ` --show-traceback`.

**Comment by ethanhs** [comment]
This is a duplicate of #708.
